samples: logger: test sample in CI

nashif opened the pull request:
Add test stub for this sample to verify it in CI on default platforms.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: logger: test sample in CI

Add test stub for this sample to verify it in CI on default platforms.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nordic-krch commented on samples/subsys/logging/logger/src/main.c:
```diff
@@ -29,6 +28,10 @@ SAMPLE_INSTANCE_DEFINE(inst1);
 #define INST2_NAME STRINGIFY(SAMPLE_INSTANCE_NAME.inst2)
 SAMPLE_INSTANCE_DEFINE(inst2);
 
+#ifndef NRF_RTC1
```
`#ifdef CONFIG_SOC_FAMILY_NRF` would be better

aescolar replied:
For #9988 it wouldn't be.

nashif commented:
recheck

nashif committed:
samples: logger: exclude qemu_xtensa [REVERTME]

The way uart console is setup for this is not compatible with the logger
and needs additional changes, excluding until we have this implemented
in a generic way.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
